src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -148,7 +157,7 @@ private long internalCheck(EC2Computer computer) {
             // * Already Terminated
             // * We use stop-on-terminate and the instance is currently stopped or stopping
             if (InstanceState.TERMINATED.equals(state)
-                  || computer.getSlaveTemplate().stopOnTerminate && (InstanceState.STOPPED.equals(state) || InstanceState.STOPPING.equals(state))) {
+                  || slaveTemplate.stopOnTerminate && (InstanceState.STOPPED.equals(state) || InstanceState.STOPPING.equals(state))) {
```

Comment:
You're absolutely right it should be there too. It wasn't initially and I just changed to re-use the variable from above and didn't think about null-checks at all. Functionally it should make no difference from today since `computer.getSlaveTemplate()` is what's used today as well. 

Nevertheless I'll add a null-check.